org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/trigger/QueueLengthTriggerChecker.java

```diff
@@ -0,0 +1,71 @@
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entity.trigger;
+
+import java.util.Set;
+
+import javax.measure.Measure;
+import javax.measure.quantity.Dimensionless;
+
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterObjectWrapper;
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterResult;
+import org.palladiosimulator.analyzer.slingshot.monitor.data.events.MeasurementMade;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPoint;
+import org.palladiosimulator.metricspec.constants.MetricDescriptionConstants;
+import org.palladiosimulator.pcmmeasuringpoint.PassiveResourceReference;
+import org.palladiosimulator.spd.triggers.SimpleFireOnValue;
+import org.palladiosimulator.spd.triggers.expectations.ExpectedCount;
+import org.palladiosimulator.spd.triggers.stimuli.QueueLength;
+
+/**
+ * The trigger checker for {@link QueueLength}. The measurements are only
+ * considered if the measuring point references a passive resource (i.e. is of
+ * type {@link PassiveResourceReference} and the referenced passive resource
+ * corresponds to the passive resource returned by
+ * {@link QueueLength#getPassiveResource()}.
+ * 
+ * @author Julijan Katic
+ *
+ */
+public class QueueLengthTriggerChecker extends TriggerChecker<QueueLength> {
+
+	private final QueueLength queueLength;
+
+	public QueueLengthTriggerChecker(final SimpleFireOnValue trigger, final QueueLength queueLength) {
+		super(trigger, QueueLength.class, Set.of(ExpectedCount.class));
+		this.queueLength = queueLength;
+	}
+
+	@Override
+	public FilterResult doProcess(final FilterObjectWrapper event) {
+		if (event.getEventToFilter() instanceof final MeasurementMade measurementMade
+				&& isCorrectPassiveResource(measurementMade.getEntity().getMeasuringPoint())) {
+
+			final Measure<Long, Dimensionless> value = measurementMade.getEntity()
```

Comment:
Can you elaborate on the first point more? I don't know why the other metrics specs make sense in this case.

Reply:
back then when i defined those monitor x metricspec. combinations for the RMUC, i did it because i can (and because i wanted to test whether the implementation for waiting and holding time works). 

the might in fact not make sense for teh RMUC, but it is still possible to define such monitor x metricspec. combinations in the monitorrepository, thus i think we should check that the measurments we get are actually a `STATE_OF_PASSIVE_RESOURCE_METRIC` and not for waiting or holding time. 

idk, did i get you question right? if not, let's haven a call and discuss verbally.

Reply:
Ah ok, I understand now. I understood that we should somehow also incoorporate measurements from the other metrics as well, but in my mind it didn't make sense.

I included a check that the measurements are actually of the `STATE_OF_PASSIVE_RESOURCE` metric